pkg/pkgupdates/pkgupdates.jsx

```diff
@@ -0,0 +1,448 @@
+    }
+
+    render() {
+        return (
+            <Tooltip tip={this.state.summary}>
+                <a href={this.props.url} rel='noopener' referrerpolicy='no-referrer' target='_blank'>{this.props.cve}</a>
+            </Tooltip>
+        );
+    }
+}
+
+// multiple packages might refer to the same CVEs, so cache/reuse the components to avoid redundant lookups
+var cveLinkCache = { };
+
+function makeCVELink(url) {
+    var cve = url.match(/[^/=]+$/);
+    if (!(cve in cveLinkCache))
+        cveLinkCache[cve] = <CVELink cve={cve} url={url}/>;
+    return cveLinkCache[cve];
+}
+
+function UpdateItem(props) {
+    const info = props.info;
+    const id_fields = props.id.split(';');
+    var bugs = null;
+    var security_info = null;
+
+    if (info.bug_urls && info.bug_urls.length)
+        // we assume a bug URL ends with a number; if not, show the complete URL
+        bugs = commaJoin(info.bug_urls.map(u => <a rel='noopener' referrerpolicy='no-referrer' target='_blank' href={u}>{u.match(/[0-9]+$/) || u}</a>));
+
+    if (info.security) {
+        security_info = (
+            <p>
+                <span className='fa fa-bug security-label'> </span>
+                <span className='security-label-text'>{_('Security Update') + (info.cve_urls.length ? ': ' : '')}</span>
+                {commaJoin(info.cve_urls.map(u => makeCVELink(u)))}
+            </p>
+        );
+    }
+
+    return (
+        <tbody>
+            <tr className={'listing-ct-item' + (info.security ? ' security' : '')}>
+                <th>
+                  <Tooltip tip={info.summary}>
+                    <span>{id_fields[0]}</span>
+                  </Tooltip>
+                </th>
+                <td className='narrow'>{id_fields[1]}</td>
+                <td className='narrow'>{bugs}</td>
+                <td className='changelog'>{security_info}{info.description}</td>
+            </tr>
+        </tbody>
+    );
+}
+
+function UpdatesList(props) {
+    // sort security first
+    var updates = Object.keys(props.updates);
+    updates.sort((a, b) => {
+        if (props.updates[a].security && !props.updates[b].security)
+            return -1;
+        if (!props.updates[a].security && props.updates[b].security)
+            return 1;
+        return a.localeCompare(b);
+    });
+
+    return (
+        <table className='listing-ct'>
+            <thead>
+                <tr>
+                    <th>{_('Name')}</th>
+                    <th>{_('Version')}</th>
+                    <th>{_('Bugs')}</th>
+                    <th>{_('Details')}</th>
+                </tr>
+            </thead>
+            {updates.map(id => <UpdateItem id={id} info={props.updates[id]} />)}
+        </table>
+    );
+}
+
+class ApplyUpdates extends React.Component {
+    constructor() {
+        super();
+        this.state = {percentage: null, timeRemaining: null, curStatus: null, curPackage: null};
+    }
+
+    componentDidMount() {
+        pkTransaction()
```

Comment:
Something like this: https://github.com/mvollmer/cockpit/commit/ac75e2ba67fe5799638b3e61a4f12926a0204db4

With this, clicking the button directly starts the transaction, and ApplyUpdates is only watching it.

I believe your approach is also correct, but one needs to understand the React component lifecycle to see how the button click leads to the wanted transaction.  That's okay reading state from the server that is needed to render a component, but it's too indirect for user initiated actions that actually modify the server.